src/components/modals/FormStepSaveModal.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Display a modal to allow the user to save the form step in it's current state.
+ */
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {useHistory} from 'react-router-dom';
+import {FormattedMessage} from 'react-intl';
+import {put, post, destroy} from 'api';
+import {getBEMClassName} from 'utils';
+import Modal from 'components/modals/Modal';
+import ErrorMessage from 'components/ErrorMessage';
+import Body from 'components/Body';
+import Label from 'components/Label';
+import Input from 'components/Input';
+import HelpText from 'components/HelpText';
+import {Toolbar, ToolbarList} from 'components/Toolbar';
+import Button from 'components/Button';
+
+
+const FormStepSaveModal = ({
+    isOpen,
+    closeModal,
+    stepData,
+    saveStepDataUrl,
+    suspendFormUrl,
+    destroySessionUrl,
+}) => {
+  const history = useHistory();
+  const [email, setEmail] = useState("");
+  const [failed, setFailed] = useState(false);
+
+  const onSubmit = async (event) => {
+    event.preventDefault();
+    let response = await put(saveStepDataUrl, {data: stepData});
+    if (!response.ok) return setFailed(true);
+    response = await post(suspendFormUrl, {email});
+    if (!response.ok) return setFailed(true);
+    try {
+      // Destroy throws an exception if the API is not successful
+      await destroy(destroySessionUrl);
+    } catch (e) {
+      return setFailed(true);
+    }
+    history.push('/');
```

Comment:
Please provide useful user feedback instead of a generic error message.

* "Saving the data failed, please try again later"
* "Suspending the form failed, please try again later"
* "Logging out failed, please try again later"

etc.